sc_cost_meter/utils.py

```diff
@@ -0,0 +1,185 @@
+  IDs to SC subscriber data
+  :param synapse_id: synapse user id
+  :return the Marketplace customer ID, otherwise return None if cannot find an
+          associated customer ID
+  '''
+  customer_id = None
+  ddb_marketplace_table_name = get_env_var_value('MARKETPLACE_ID_DYNAMO_TABLE_NAME')
+  if ddb_marketplace_table_name:
+    ddb_customer_id_attribute = 'MarketplaceCustomerId'
+    client = get_dynamo_client()
+    response = client.get_item(
+      Key={
+        'SynapseUserId': {
+          'S': synapse_id,
+        }
+      },
+      TableName=ddb_marketplace_table_name,
+      ConsistentRead=True,
+      AttributesToGet=[
+        ddb_customer_id_attribute
+      ]
+    )
+
+    if "Item" in response.keys():
+      customer_id = response["Item"][ddb_customer_id_attribute]["S"]
+    else:
+      log.info(f'cannot find registration for synapse user: {synapse_id}')
+
+  return customer_id
+
+def get_marketplace_product_code(synapse_id):
+  '''Get the registered Service Catalog customer product code.
+  Assumes that there is a Dynamo DB with a table containing a mapping of Synapse
+  IDs to SC subscriber data
+  :param synapse_id: synapse user id
+  :return the Marketplace product code, None if cannot find customer ID
+  '''
+  product_code = None
+  ddb_marketplace_table_name = get_env_var_value('MARKETPLACE_ID_DYNAMO_TABLE_NAME')
+  if ddb_marketplace_table_name:
+    ddb_product_code_attribute = 'ProductCode'
+    client = get_dynamo_client()
+    response = client.get_item(
+      Key={
+        'SynapseUserId': {
+          'S': synapse_id,
+        }
+      },
+      TableName=ddb_marketplace_table_name,
+      ConsistentRead=True,
+      AttributesToGet=[
+        ddb_product_code_attribute
+      ]
+    )
+
+    if "Item" in response.keys():
+      product_code = response["Item"][ddb_product_code_attribute]["S"]
+      log.debug(f'marketplace product code: {product_code}')
+    else:
+      log.info(f'cannot find registration for synapse user: {synapse_id}')
+
+  return product_code
+
+
+def get_customer_cost(customer_id, time_period, granularity):
+  '''
+  Get the total cost of all resources tagged with the customer_id for a given
+  time_period.  The time_period and time granularity must match.
+  :param customer_id: the Marketplace customer ID
+  :param time_period: the cost time period
+  :param granularity: the granularity of time HOURLY|DAILY|MONTHLY
+  :return: the total cost of all resources and the currency unit
+  '''
+  client = get_ce_client()
+
+  response = client.get_cost_and_usage(
+    TimePeriod=time_period,
+    Granularity=granularity,
+    Filter={
+      "Tags": {
+        "Key": "marketplace:customerId",
+        "Values": [
+          customer_id
+        ]
+      }
+    },
+    Metrics=["UnblendedCost"]
+  )
+
+  results_by_time = response['ResultsByTime']
+  cost = results_by_time[0]["Total"]["UnblendedCost"]["Amount"]
+  unit = results_by_time[0]["Total"]["UnblendedCost"]["Unit"]
```

Comment:
Is it guaranteed that 'unit' is always US Dollars?

Reply:
Units can be for resource usage or cost.  In this specific request we are asking for the cost so we are at least guaranteed that the unit is a currency.  Now if you asking whether this will always return the currency then i'm not sure.  I don't see anything that allows you to get something other than US dollars from `get_cost_and_usage` API.  I'm guessing that it must depend on your AWS account's currency setting?  https://aws.amazon.com/blogs/aws/new-set-preferred-payment-currency-for-your-aws-account/